### PR TITLE
Moved BaseHolder and Source to core ee

### PR DESCRIPTION
**Pull request by gregw**
This was a lot of work just to move 2 small classes to core.
We have to evaluate if the reduced duplication is worth the complexity of the split.

Note that I have not yet updated the EE10 code to use these common classes, will be the same changes as made to EE11

### CODE REVIEW

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHolder.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+
+/**
+ * Specialization of BaseHolder for servlet-related classes
+ *
+ * @param <T> the type of holder
+ */
+@ManagedObject("Holder - a container for servlets and the like")
+public abstract class ServletContextHolder<T> extends BaseHolder<T>
```
Comment (janbartel) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHolder.java:30]: I think this name is misleading: it is not holding a `ServletContext`. Not sure of what a better name might be? `ServletContextScopedHolder`?

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHolder.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            setContextHandler(_servletHandler.getServletContextHandler());
+    }
+
+    @Override
+    protected Exception newUnavailableException(String message)
+    {
+        return new UnavailableException(message);
+    }
+
+    @ManagedAttribute(value = "Display Name", readonly = true)
```
Comment (janbartel) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHolder.java:93]: Can't all the methods from here on down be pushed down into `BaseHolder`? They're independent of the servlet spec version, and I don't think its important if they are not used by some of the derived classes.

**Review by janbartel — CHANGES_REQUESTED** [review]
I'm in 2 minds about this: one the one hand it is nice to reduce code duplication, plus a lot of the changes are just mechanical and not substantive; but on the other hand it is a  PITA.

**Review by lorban — COMMENTED** [review]
My feeling is -1 on this change.

IMHO the cost of this rather small code deduplication isn't worth the complexity of this change.

**Comment by gregw** [comment]
I'll close this soon, but I'll first look through for any worthwhile cleanups in the PR.

**Pull request closed without merging**